app/scripts/controllers/sort-network-list.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  BaseControllerV2,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { NetworkState } from '@metamask/network-controller';
+
+const controllerName = 'NetworksOrderingController';
+
+/**
+ * The network ID of a network.
+ */
+export type NetworkId = string;
+
+export type NetworksOrderingControllerState = {
+  networksList: NetworkId[] | null;
+};
+
+export type NetworksOrderingControllerupdateNetworksListAction = {
+  type: `${typeof controllerName}:updateNetworksList`;
+  handler: NetworksOrderingController['updateNetworksList'];
+};
+
+export type NetworksOrderingControllerMessengerActions =
+  NetworksOrderingControllerupdateNetworksListAction;
+
+export type NetworksOrderingControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  NetworksOrderingControllerMessengerActions,
+  never,
+  never,
+  never
+>;
+
+const defaultState = {
+  networksList: [],
+};
+
+const metadata = {
+  networksList: {
+    persist: true,
+    anonymous: true,
+  },
+};
+
+export class NetworksOrderingController extends BaseControllerV2<
+  typeof controllerName,
+  NetworksOrderingControllerState,
+  NetworksOrderingControllerMessenger
+> {
+  /**
+   * Creates a NetworksOrderingController instance.
+   *
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
+   * @param args.onNetworkStateChange - Allows subscribing to network controller state changes.
+   */
+  constructor({
+    messenger,
+    state,
+    onNetworkStateChange,
+  }: {
+    messenger: NetworksOrderingControllerMessenger;
+    state?: NetworksOrderingControllerState;
+    onNetworkStateChange: (listener: (state: NetworkState) => void) => void;
+  }) {
+    super({
+      messenger,
+      metadata,
+      name: controllerName,
+      state: { ...defaultState, ...state },
+    });
+
+    if (onNetworkStateChange) {
```

Comment:
This branch can be removed; we had used `onNetworkStateChange` listeners for tracking network changes back when the old extension network controller was used (it didn't use the messaging system). But now that the network controllers have been merged, we can just rely on the `NetworkController:stateChange` event and drop this redundant event listener

Reply:
Hmm.. If I remove this condition and just rely on `messagingSystem.subscribe` , extension doesn't seem to load